python/pyspark/sql/functions/builtin.py

```diff
@@ -18787,7 +18771,7 @@ def exists(col: "ColumnOrName", f: Callable[[Column], Column]) -> Column:
     |        true|
     +------------+
     """
-    return _invoke_higher_order_function("ArrayExists", [col], [f])
+    return _invoke_higher_order_function("exists", [col], [f])
```

Comment:
Probably I am overthinking but just for the sake of review, this will now call whatever `exists` is registered as, e.g., users could replace the existing function via `spark.register`.

Reply:
That is also a problem for all other functions in here.

Reply:
...and all functions used on the scala side...

We can put it in a different namespace (e.g. builtin) - much like the internal thing we recently added - but that is for a different PR.